homeassistant/components/hassio/handler.py

```diff
@@ -332,7 +322,16 @@ def __init__(
         self.loop = loop
         self.websession = websession
         self._ip = ip
-        self._base_url = URL(f"http://{ip}")
+        base_url = f"http://{ip}"
+        self._base_url = URL(base_url)
+        self._client = SupervisorClient(
+            base_url, os.environ.get("SUPERVISOR_TOKEN", ""), session=websession
+        )
+
+    @property
+    def client(self) -> SupervisorClient:
+        """Return aiohasupervisor client."""
+        return self._client
```

Comment:
```suggestion
        self.client = SupervisorClient(
            base_url, os.environ.get("SUPERVISOR_TOKEN", ""), session=websession
        )
```
Why make it protected if you're going to expose it anyway :)

Reply:
Because if you do it that way then someone can do this:
```py
hassio = HassIO(...)
hassio.client = SupervisorClient(...)
```
By making it a property like that `client` is read-only on instances of `HassIO`. Others can access it to make calls to supervisor, they cannot change its value to something new.

Reply:
I am not aware if we have places in the code where we explictly do this, let me check with others

Reply:
Okay, talked with Martin and this would be good in its current form